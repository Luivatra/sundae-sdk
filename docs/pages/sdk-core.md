<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@sundae/sdk-core](./sdk-core.md)

## sdk-core package

## Classes

|  Class | Description |
|  --- | --- |
|  [AssetAmount](./sdk-core.assetamount.md) |  |
|  [Provider](./sdk-core.provider.md) |  |
|  [ProviderSundaeSwap](./sdk-core.providersundaeswap.md) |  |
|  [SundaeSDK](./sdk-core.sundaesdk.md) |  |
|  [SwapConfig](./sdk-core.swapconfig.md) |  |
|  [TxBuilder](./sdk-core.txbuilder.md) |  |
|  [TxBuilderLucid](./sdk-core.txbuilderlucid.md) |  |

## Enumerations

|  Enumeration | Description |
|  --- | --- |
|  [ESupportedWallets](./sdk-core.esupportedwallets.md) |  |

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [IAsset](./sdk-core.iasset.md) |  |
|  [IAssetMetadata](./sdk-core.iassetmetadata.md) |  |
|  [IAssetParsedID](./sdk-core.iassetparsedid.md) |  |
|  [IBuildSwapArgs](./sdk-core.ibuildswapargs.md) |  |
|  [IGetSwapArgs](./sdk-core.igetswapargs.md) |  |
|  [IParams](./sdk-core.iparams.md) |  |
|  [IPoolData](./sdk-core.ipooldata.md) |  |
|  [IPoolDataAsset](./sdk-core.ipooldataasset.md) |  |
|  [IPoolQuery](./sdk-core.ipoolquery.md) |  |
|  [ISwapResponse](./sdk-core.iswapresponse.md) |  |
|  [ITxBuilderLoaderOptions](./sdk-core.itxbuilderloaderoptions.md) |  |

## Type Aliases

|  Type Alias | Description |
|  --- | --- |
|  [TDatumType](./sdk-core.tdatumtype.md) |  |
|  [TLucidArgs](./sdk-core.tlucidargs.md) |  |
|  [TMeshArgs](./sdk-core.tmeshargs.md) |  |
|  [TSupportedNetworks](./sdk-core.tsupportednetworks.md) |  |
|  [TSupportedTxBuilderLibs](./sdk-core.tsupportedtxbuilderlibs.md) |  |
|  [TSupportedTxBuilderOptions](./sdk-core.tsupportedtxbuilderoptions.md) |  |
|  [TSupportedTxBuilders](./sdk-core.tsupportedtxbuilders.md) |  |
|  [TSupportedTxBuilderTxTypes](./sdk-core.tsupportedtxbuildertxtypes.md) |  |
|  [TSwapAsset](./sdk-core.tswapasset.md) |  |
|  [TTxBuilderComplete](./sdk-core.ttxbuildercomplete.md) |  |

