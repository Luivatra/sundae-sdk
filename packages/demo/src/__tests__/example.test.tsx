import { Mocks } from "@sundaeswap/sdk-core/testing";
Mocks.MockExports();

import "@testing-library/jest-dom";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { AppStateProvider } from "../state/context";
import Actions from "../components/Actions";
import { SundaeSDK } from "@sundaeswap/sdk-core";
import {
  ProviderSundaeSwap,
  TxBuilderLucid,
} from "@sundaeswap/sdk-core/extensions";

console.log = jest.fn();

describe("Example testing", () => {
  it("should use the mocked SundaeSDK", async () => {
    const MockedSDK = new SundaeSDK(
      new TxBuilderLucid(
        {
          network: "preview",
          wallet: "eternl",
          provider: "blockfrost",
          blockfrost: {
            apiKey: "",
            url: "",
          },
        },
        new ProviderSundaeSwap("preview")
      )
    );

    const { container, getByText } = render(
      <AppStateProvider defaultValue={{ SDK: MockedSDK }}>
        <Actions />
      </AppStateProvider>
    );

    const swapButton = getByText("Swap tINDY for tADA");
    expect(swapButton).toBeInTheDocument();

    fireEvent.click(swapButton);
    await waitFor(() => {
      expect(Mocks.SundaeSDK.mockQuery).toHaveBeenCalledTimes(1);
      expect(Mocks.SundaeSDK.mockSwap).toHaveBeenCalledTimes(1);
      expect(console.log).toHaveBeenCalledTimes(2);
    });

    expect(container).toMatchSnapshot();
  });
});
