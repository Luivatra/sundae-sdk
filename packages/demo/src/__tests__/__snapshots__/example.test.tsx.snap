// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example testing should use the mocked SundaeSDK 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <h2
      class="mb-4 text-lg font-bold text-white"
    >
      SundaeSwap Protocol Actions
    </h2>
    <div
      class="grid grid-cols-3 gap-4"
    >
      <button
        class="flex items-center justify-center gap-4 rounded-md bg-blue-500 px-4 py-2 font-bold text-white transition-colors hover:bg-blue-800"
      >
        Swap tINDY for tADA
      </button>
      <button
        class="flex items-center justify-center gap-4 rounded-md bg-blue-500 px-4 py-2 font-bold text-white transition-colors hover:bg-blue-800"
      >
        Swap tADA for tINDY
      </button>
    </div>
  </div>
</div>
`;
